amortization.py

```diff
@@ -231,8 +254,38 @@ def __str__(self):
                    (self.rate, self.nper, self.pv, self.typ))
 
 
+    def _calc_pmt_diff_(self, pmt):
+        """ Difference between last payment and penultimate payment """
+        l = Loan(self.rate, self.nper, self.pv, self.date, payment=pmt)
+        return l.period(l.nper).payment - l.period(l.nper-1).payment
+
     def pmt(self):
-        return self.payment or pmt(self.rate, self.nper, self.pv, self.typ)
+        if self.payment:
+            return self.payment
+
+        if PMT_METHOD == ORDINARY_ANNUITY_PMT_METHOD:
+            return pmt(self.rate, self.nper, self.pv, self.typ)
+        else: # NEWTONS_METHOD
+            # This isn't actually Newton's method, which would use derivatives of our
+            # function, but approximates the derivative by just dividing the difference by number of periods
+            start = pmt(self.rate, self.nper, self.pv, self.typ)
+            diff = self._calc_pmt_diff_(start)
+            iterations = 0
+            while iterations < 50 and abs(diff * 100) > self.nper:
+                # iterate until payment would change by less than 0.01 or
+                # until 50 iterations.
+                iterations += 1
+                start += diff / self.nper
```

Comment:
It may help to name `start` as `starting_payment` to make sure there is no confusing it with a starting date.